### DISCUSSION TITLE
How do I get the local candidate and send to remote?

### DISCUSSION THREAD

**Discussion by wdai9162** [aiortc/aiortc#675] [Q&A]
Im a beginner on WebRTC and especially this package, please excuse me if this is a dumb question. 
In JS implementation, both ends need to send local `icecandidate` to each other so that `addIceCandidate`(remote) can be called. 

For JS implementation, when [RTCPeerConnection.setLocalDescription()](https://developer.mozilla.org/en-US/docs/Web/API/RTCPeerConnection/setLocalDescription) is set an `icecandidate` event will be fired and I can send the local `icecandidate` info to the remote end. 

Since there seems no such event in this package, I have been stuck and can't find a way to retrieve local `icecandidate` info. I don't know when it gets generated nor where to find it...Consequently, I can't provide the candidate information to remote end and connection fails. 

It will be appreciated if anyone can shed some light. Thanks in advance.

**Comment by tristan-f-r** [comment]
For those who are confused, [a similar problem is described here](https://stackoverflow.com/questions/28450738/onicegatheringstatechange-event-does-not-trigger-during-candidates-gathering) in JS. The author is looking for a similar event that would fire when an ice candidate is added to enable ICE trickling.

Unfortunately, @wdai9162 , ICE trickling on the aiortc side is [still being implemented](https://github.com/aiortc/aioice/issues/4) (only a half-implementation of aiortc ice trickling is done)
